client-js/app/elements/labrad-registry.html

```diff
@@ -59,8 +207,213 @@
       path: {
         type: Array,
         notify: true
+      },
+      socket: {
+        type: Object,
+        notify: true,
+      },
+      selectType: {
+        type: String,
+        notify: true
       }
-    }
+    },
+    ready: function() {
+      console.log(Polymer.version);
+      this.addEventListener('iron-form-submit',this.updateKey);
+    },
+    //Helper Functions
+    selectKey: function() {
+      this.selDir = null;
+      this.selectType = 'key';
+    },
+    selectDir: function(){
+      this.selKey = null;
+      this.selectType = 'dir';
+    },
+    incrementSelector: function() {
+      console.log(this.selKey);
+      //TODO increment selected key on tab
+    },
+    repopulateList: function(resp) {
+      this.selDir=null;
+      this.splice('dirs',0,this.dirs.length);
+      this.splice('keys',0,this.keys.length);
+
+      for (var i in resp.dirs) {
+        //bw = resp.dirs.length-1-i; //run through list backwards so it orders correctly
+        //console.log(resp.dirs[i]);
+        this.push('dirs',{name: resp.dirs[i], url: this.createUrl(resp.path,resp.dirs[i])});
+      }
+      for (var j in resp.keys){ 
+        //bw = resp.keys.length-1-i; //run through list backwards so it orders correctly
+        this.push('keys',{name: resp.keys[j], value: resp.vals[j]});
+      }
+    },
+    createUrl: function(path,dir) {
+      var pathUrl = '/registry/';
+      if (path.length === 0){
+        return pathUrl+dir;
+      }//not sure if this is the best way to handle this edge case
+      for (var i in path) {
+        pathUrl += path[i] + '/';
+      }
+      console.log(pathUrl+dir);
+      return pathUrl+dir;
+    },
+    pathToString: function(path) {
+      return JSON.stringify(path);
+    },
+    handleError: function(error) {
+      //we can add a more creative way of displaying errors here
+      console.log(error);
+    },
+    /////////////////////
+    //Interface Functions
+    ////////////////////
+    doDelete: function() {
+      var self = this;
+      var p1;
+      
+      if (self.selectType==='dir') {
+        p1 = self.socket.rmDir({path: self.path,dir: self.selDir});
+        p1.then(function(resp) {
+          self.repopulateList(resp);
+        }, 
+        function(reason) {
+          self.handleError(reason); 
+        });
```

Comment:
the indentation here is a bit misleading, making it unclear that this second function is the error callback. I would suggest something like:
```javascript
p1.then(
  function(resp) { self.repopulateList(resp); },
  function(reason) { self.handleError(reason); }
);
```